redisvl/index/index.py

```diff
@@ -476,6 +477,26 @@ def delete(self, drop: bool = True):
         except:
             logger.exception("Error while deleting index")
 
+    def clear(self) -> int:
+        """Clear all keys in Redis associated with the index, leaving the index
+        available and in-place for future insertions or updates.
+
+        Returns:
+            int: Count of records deleted from Redis.
+        """
+        # Track deleted records
+        total_records_deleted: int = 0
+
+        # Paginate using queries and delete in batches
+        for batch in self.paginate(
+            FilterQuery(FilterExpression("*"), return_fields=["id"]), page_size=500
```

Comment:
Unlikely scenario, but what if I want to get the (some) data out before destroying the index, e.g. `return_fields` being configurable also?

Reply:
I agree with the notion that someone might want to be able to export data from an index. However, I think that should be it's own clear feature. For example, an `index.export(file_path="file.json")` method or something similar?

I think `clear()` should just be a simple method for clearing data out of your index